sql/expression/function/math_test.go

```diff
@@ -78,4 +80,207 @@ func TestRandWithSeed(t *testing.T) {
 	f642 = f.(float64)
 
 	assert.Equal(t, f64, f642)
-}
\ No newline at end of file
+}
+
+func TestAbsValue(t *testing.T) {
+	type toTypeFunc func(float64) interface{}
+
+	decimal1616 := sql.MustCreateDecimalType(16,16)
+
+	toInt64 := func(x float64) interface{} { return int64(x) }
+	toInt32 := func(x float64) interface{} { return int32(x) }
+	toInt := func(x float64) interface{} { return int(x) }
+	toInt16 := func(x float64) interface{} { return int16(x) }
+	toInt8 := func(x float64) interface{} { return int8(x) }
+	toUint64 := func(x float64) interface{} { return uint64(x) }
+	toUint32 := func(x float64) interface{} { return uint32(x) }
+	toUint := func(x float64) interface{} { return uint(x) }
+	toUint16 := func(x float64) interface{} { return uint16(x) }
+	toUint8 := func(x float64) interface{} { return uint8(x) }
+	toFloat64 := func(x float64) interface{} {return x}
+	toFloat32 := func(x float64) interface{} {return float32(x)}
+	toDecimal1616 := func(x float64) interface{} {return decimal.NewFromFloat(x)}
+
+	signedTypes := map[sql.Type]toTypeFunc {
+		sql.Int64: toInt64,
+		sql.Int32: toInt32,
+		sql.Int24: toInt,
+		sql.Int16: toInt16,
+		sql.Int8: toInt8}
+	unsignedTypes := map[sql.Type]toTypeFunc {
+		sql.Uint64: toUint64,
+		sql.Uint32: toUint32,
+		sql.Uint24: toUint,
+		sql.Uint16: toUint16,
+		sql.Uint8: toUint8}
+	floatTypes := map[sql.Type]toTypeFunc{
+		sql.Float64: toFloat64,
+		sql.Float32: toFloat32,
+		decimal1616: toDecimal1616,
+	}
+	expectedConv := map[sql.Type]toTypeFunc{
+		sql.Int64: toInt64,
+		sql.Int32: toInt64,
+		sql.Int24: toInt64,
+		sql.Int16: toInt64,
+		sql.Int8: toInt64,
+		sql.Uint64: toUint64,
+		sql.Uint32: toUint64,
+		sql.Uint24: toUint64,
+		sql.Uint16: toUint64,
+		sql.Uint8: toUint64,
+		sql.Float64: toFloat64,
+		sql.Float32: toFloat64,
+		decimal1616: toDecimal1616,
+	}
+
+	testCases := []struct {
+		name     string
+		typeToConv    map[sql.Type]toTypeFunc
+		val      float64
+		expected float64
+		err      error
+	}{
+		{
+			"signed types positive int",
+			signedTypes,
+			5.0,
+			5.0,
+			nil,
+		},{
+			"signed types negative int",
+			signedTypes,
+			-5.0,
+			5.0,
+			nil,
+		},
+		{
+			"unsigned types positive int",
+			unsignedTypes,
+			5.0,
+			5.0,
+			nil,
+		},
+		{
+			"float positive int",
+			floatTypes,
+			5.0,
+			5.0,
+			nil,
+		},{
+			"float negative int",
+			floatTypes,
+			-5.0,
+			5.0,
+			nil,
+		},
+	}
+
+	newAbsVal := NewUnaryMathFunc("abs", AbsFuncLogic{})
```

Comment:
You should declare constructor funcs for every new func you added

Reply:
Why?  The reason I went to the trouble of building this abstraction was so that I didn't have to write a bunch of useless shit and could just write the functionality.  The constructor is empty

Reply:
Just seems like a leaky abstraction to me. Sharing implementation details is good, exporting that detail to clients is bad.

Reply:
I don't think go shares your philosophy on this.  Go doesn't even provide real constructors, as a convention we often use constructor like factory methods when the zero value of a class needs additional initialization.  In this case that isn't needed and the zero value is used without any need to write:

```
func NewAbsFuncLogic() {
    return AbsFuncLogic{}
}
```